src/Middleware/RateLimiting/src/DefaultKeyTypeEqualityComparer.cs

```diff
@@ -0,0 +1,30 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.AspNetCore.RateLimiting;
+
+internal sealed class DefaultKeyTypeEqualityComparer : IEqualityComparer<DefaultKeyType>
+{
+    public bool Equals(DefaultKeyType x, DefaultKeyType y)
+    {
+        var xKey = x.Key;
+        var yKey = y.Key;
+        if (xKey == null && yKey == null)
+        {
+            return x.PolicyName.Equals(y.PolicyName);
```

Comment:
Nit:

```suggestion
            return string.Equals(x.PolicyName, y.PolicyName, StringComparison.Ordinal)
```

Or if you prefer to be less explicit:

```suggestion
            return x.PolicyName == y.PolicyName;
```

They're equivalent. Same for the final return.